### ISSUE TITLE
iOS sample app crashes when API token not provided

### ISSUE THREAD

**Issue by akitchen** [mapbox/mapbox-gl-native#10768]
**Platform:** iOS

**Mapbox SDK version:** 3.7.1

### Steps to trigger behavior

 1. Build & run the iosapp target without a token set in the Info.plist or environment

### Expected behavior

App displays an alert view prompting the user to paste in their token

### Actual behavior

App prompts the user to paste in their token but also crashes with the following written to the logs (user may or may not see the alert view due to timing):
```
2017-12-20 13:36:12.956701-0800 Mapbox GL[49405:7128021] [ERROR] {}[Setup]: loading style failed: HTTP status code 401
libc++abi.dylib: terminating with uncaught exception of type std::runtime_error: You must provide a Mapbox API access token for Mapbox tile sources
```

**Comment by friedbunny** [comment]
Have you enabled the new build system in Xcode 9.x?

**Comment by 1ec5** [comment]
As with #10641, this issue is caused by `MGLDocumentationExampleTests.testMGLMapSnapshotter()` calling an asynchronous method without waiting on it to return.

**Comment by akitchen** [comment]
This has been fixed both in 3.7.x and on master.
